### ISSUE TITLE
Message for raise-missing-from

### ISSUE THREAD

**Issue by cool-RR** [PyCQA/pylint#3707]
I think that the message we added to the `raise-missing-from` rule isn't user friendly enough: https://github.com/PyCQA/pylint/blob/master/pylint/checkers/exceptions.py#L157

The goal is that a user who has never seen the syntax before will be able to use it. We might also want to link to PEP 3134, hopefully to the relevant segment.

Is anyone interested in writing the new message? Otherwise I'll give it a try.

**Comment by cool-RR** [comment]
Here's my suggestion for the title:

```
Exception cause needed: raise new_exception from old_exception
```

And for the content:

```
You are raising the %s exception after the %s exception was caught. It's likely that the second exception is a friendly re-wrapping of the first exception, therefore you should use `raise new_exception from old_exception` to see an accurate message between the two tracebacks. See PEP 3132 for more information: https://www.python.org/dev/peps/pep-3132/
```

What do you think?

**Comment by adelton** [comment]
The link should point to https://www.python.org/dev/peps/pep-3134/, not https://www.python.org/dev/peps/pep-3132/.

**Comment by AWhetter** [comment]
Sounds like a good idea to me!

**Comment by jnsnow** [comment]
I might also mention that "from None" is a viable solution when you DON'T want to re-surface the inner context. I think this is valid in situations like when you are transforming KeyError into a domain-appropriate error with a semantic meaning. The KeyError itself is almost entirely uninteresting and just adds lines to the console buffer.

Another note is that Exception Chaining has not been mentioned in the Python tutorial ... until 3.9, which is not yet released. I would argue that this technique is not in wide circulation as a best practice yet. After 3.9 finalizes, I suggest linking to the tutorial instead of the PEP, which is going to be easier to follow and understand for most people: https://docs.python.org/3.9/tutorial/errors.html#exception-chaining

**Comment by imbaczek** [comment]
i'd like to point out that making users add 'from None' to their raises just to silence warnings brings us closer to Python 2 where useful debug info was hidden when raising in exception handlers and you had to jump through hoops to not lose tracebacks.

**Comment by cool-RR** [comment]
> i'd like to point out that making users add 'from None' to their raises just to silence warnings brings us closer to Python 2 where useful debug info was hidden when raising in exception handlers and you had to jump through hoops to not lose tracebacks.

Absolutely agreed.

**Comment by dangayle** [comment]
In my case, it is showing this warning even though I am purposely raising a different error:

<img width="1044" alt="Screen Shot 2021-03-18 at 12 14 03 AM" src="https://user-images.githubusercontent.com/129611/111587026-e3a80300-877e-11eb-9aba-335a2670846c.png">

I came to this bug because I was searching for a way to disable this warning.

**Comment by cool-RR** [comment]
You should change `except NoResultFound:` to `except NoResultFound as no_result_found:` and later `raise NoZoomOccurrences('...') from no_result_found`.

The `NoZoomOccurrences` error didn't just happen in the context of `NoResultFound`. It's directly caused by it. 

By the way, nice usage of the walrus operator :)

**Comment by cool-RR** [comment]
Also, this is not related to this ticket.

**Comment by dangayle** [comment]
It's related in that I didn't know what the warning was really about, which
is what the ticket is about. If you wanted proof that the warning could be
more useful, consider this exhibit A :)

**Comment by cool-RR** [comment]
Thank you Pierre!
